engine/src/main/java/org/terasology/rendering/openvrprovider/OpenVRProviderSingleton.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.openvrprovider;
+
+/**
+ * Created by john on 1/9/17.
+ */
+public class OpenVRProviderSingleton {
+    public static OpenVRProvider vrProvider = new OpenVRProvider();
+}
```

Comment:
Hmmm... I know what you are trying to do but this isn't a singleton. Anybody with access to OpenBRProviderSingleton can replace the content of vrProvider with a new provider!

I'm guessing you could mitigate this by adding the final keyword but a minimally modified version of the classical singleton java pattern is this:

```
public final class ClassicSingleton {
   private static ClassicSingleton instance = null;
   private ClassicSingleton() {
      // Exists only to defeat instantiation.
   }
   public static ClassicSingleton getInstance() {
      if(instance == null) {
         instance = new ClassicSingleton();
      }
      return instance;
   }
}
```

That been said, can't you make OpenVRProvider a singleton of it own?